src/scripts/main.js

```diff
@@ -1,3 +1,47 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.target.closest('.logo')) {
+      resolve('Promise was resolved!');
+    }
+  });
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('Promise was rejected!'));
+  }, 3000);
+});
```

Comment:
The promise `promise2` is set to reject after a 3-second timeout. This is correct, but ensure that the rejection is handled properly in the `.catch` block.